### DISCUSSION TITLE
Change Python to Golang

### DISCUSSION THREAD

**Discussion by gawsoftpl** [treeverse/dvc#9010] [General]
Do you plan to change python for golang? 

Install DVC in the pipeline with other dependencies is insane. Using python in devops requirements is wrong direction.

**Comment by shcheklein** [comment]
We don't have a short term plan to change the language. Even if that improves the Ops part of it it would complicate the integration into Python DS/ML ecosystem most likely and overall it would be a huge undertaking. We are considering rewriting some core functionality into C/Rust/whatever for performance reasons.

I see you pain @gawsoftpl and also not a big fan of Python deps and the overall ecosystem tbh. It's fragile, is constantly breaking, etc. Saying that, If we talk about DVC distribution, would self-contained binary distribution work for you? DVC has `.deb`, `pyinstaller`, etc, etc. I think one can also utilize also `pex` for example to package DVC and deps into one executable. It worked for me with Spark.

**Reply by petebachant** [reply]
Are those thoughts documented anywhere? I'd potentially be up for rewriting a subcomponent bottleneck in a faster language, but I'd be curious if you already know which would be the biggest bang for the buck.

As for needing to install DVC with other dependencies, `uv tool install dvc` or simply `uvx dvc` are nice options to avoid that, so long as you don't need to use the DVC Python API.

**Comment by skshetry** [comment]
Also see https://github.com/iterative/dvc/discussions/6547 for previous discussion.
